thorlcr/activities/funnel/thfunnelslave.cpp

```diff
@@ -309,20 +312,46 @@ class FunnelSlaveActivity : public CSlaveActivity
                 unsigned i;
                 for (i=0; i<inputs.ordinality(); i++) eog[i] = false;
             }
-            current = NULL;
             currentMarker = 0;
             readThisInput = 0;
-            ForEachItemIn(i, inputs)
+
+            auto startInputNFunc = [&](unsigned i)
             {
                 try { startInput(i); }
                 catch (CATCHALL)
                 {
                     ActPrintLog("FUNNEL(%" ACTPF "d): Error staring input %d", container.queryId(), i);
                     throw;
                 }
-                if (!current)
-                    current = queryInputStream(i);
+            };
+
+            std::vector<std::future<void>> results;
+            ForEachItemIn(i, inputs)
+            {
+                results.push_back(std::async(std::launch::async, startInputNFunc, i));
```

Comment:
I assume you used this rather than CAsyncFor because it was simpler to code?

@richardkchapman  The code looks fine, but worth discussing what our policy is before situations like this?  Should we be using std::async more?

Reply:
>I assume you used this rather than CAsyncFor because it was simpler to code?

A CAsyncFor may be more concise. 
But the use of a lambda function is a clearer pattern, than deriving a class to do this.

@ghalliday @richardkchapman , change it to use a derived CAsyncFor instance?